CMakeLists.txt

```diff
@@ -3,46 +3,88 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 # cmake-format: on
 
-cmake_minimum_required(VERSION 3.23)
+set(CMAKE_SKIP_TEST_ALL_DEPENDENCY FALSE)
 
-option(BUILD_TESTING "Build tests" ON)
+cmake_minimum_required(VERSION 3.25...3.31)
 
 project(
-    beman.inplace_vector
+    beman_inplace_vector
```

Comment:
This does not conform to our naming tradition, why changing it to `beman_inplace_vector`?

We have used `beman.short_name` for almost all other projects.

[Exemplar](https://github.com/beman-project/exemplar/blob/849f0bdd39c189c6ca6e05544fb89d9b292d6b6e/CMakeLists.txt#L5-L10), [utf_view](https://github.com/beman-project/utf_view/blob/main/CMakeLists.txt#L9), [iterator_interface](https://github.com/beman-project/iterator_interface/blob/main/CMakeLists.txt#L8-L11),

Reply:
I changed it because I needed  it for `cmake export config package`

And it was used like this in `excecuter26` project.

Reply:
That will end in a very strange export package, not very common:
```bash
bash-5.2$ tree /usr/local/lib/cmake/boost_
boost_atomic-1.86.0/                    boost_graph_parallel-1.86.0/            boost_math_tr1l-1.86.0/                 boost_system-1.86.0/
boost_charconv-1.86.0/                  boost_headers-1.86.0/                   boost_nowide-1.86.0/                    boost_test_exec_monitor-1.86.0/
boost_chrono-1.86.0/                    boost_iostreams-1.86.0/                 boost_prg_exec_monitor-1.86.0/          boost_thread-1.86.0/
boost_container-1.86.0/                 boost_json-1.86.0/                      boost_process-1.86.0/                   boost_timer-1.86.0/
boost_context-1.86.0/                   boost_locale-1.86.0/                    boost_program_options-1.86.0/           boost_type_erasure-1.86.0/
boost_contract-1.86.0/                  boost_log-1.86.0/                       boost_random-1.86.0/                    boost_unit_test_framework-1.86.0/
boost_coroutine-1.86.0/                 boost_log_setup-1.86.0/                 boost_regex-1.86.0/                     boost_url-1.86.0/
boost_date_time-1.86.0/                 boost_math_c99-1.86.0/                  boost_serialization-1.86.0/             boost_wave-1.86.0/
boost_exception-1.86.0/                 boost_math_c99f-1.86.0/                 boost_stacktrace_addr2line-1.86.0/      boost_wserialization-1.86.0/
boost_fiber-1.86.0/                     boost_math_c99l-1.86.0/                 boost_stacktrace_basic-1.86.0/          
boost_filesystem-1.86.0/                boost_math_tr1-1.86.0/                  boost_stacktrace_from_exception-1.86.0/ 
boost_graph-1.86.0/                     boost_math_tr1f-1.86.0/                 boost_stacktrace_noop-1.86.0/           
bash-5.2$ tree /usr/local/lib/cmake/boost_atomic-1.86.0
/usr/local/lib/cmake/boost_atomic-1.86.0
|-- boost_atomic-config-version.cmake
|-- boost_atomic-config.cmake
|-- libboost_atomic-variant-mt-shared.cmake
`-- libboost_atomic-variant-mt-static.cmake

1 directory, 4 files
bash-5.2$ tree stagedir/lib/cmake/
stagedir/lib/cmake/
`-- beman_inplace_vector
    |-- beman_inplace_vector-config-version.cmake
    |-- beman_inplace_vector-config.cmake
    `-- beman_inplace_vector-targets.cmake

2 directories, 3 files
bash-5.2$ 
```

Reply:
and it may be used like this:
```cmake
#---------------------------------------------------------------------------------------
# search required packages and libs
#---------------------------------------------------------------------------------------

find_package(
    Boost 1.86 CONFIG
    COMPONENTS filesystem asio headers
    REQUIRED
)
```

Reply:
I would also agree that using `_` is more idiomatic than `.` within a project/target name.

Reply:
### The most important point in this **PR**!

So you agree this change?

Reply:
Hey, according to: https://github.com/bemanproject/beman/pull/69

We cannot update the project/ target name.